test/libraries/SafeToL2Setup.spec.ts

```diff
@@ -0,0 +1,245 @@
+                // Hardhat uses the most basic struct/opcode logger tracer: https://geth.ethereum.org/docs/developers/evm-tracing/built-in-tracers#struct-opcode-logger
+                // To find the "snapshot" of the storage before the DELEGATECALL into the library, we need to find the first DELEGATECALL opcode calling into the library
+                // To do that, we search for the DELEGATECALL opcode with the stack input pointing to the library address
+                const delegateCallIntoTheLib = trace.structLogs.findIndex(
+                    (log) =>
+                        log.op === "DELEGATECALL" &&
+                        sameHexString(log.stack[log.stack.length - 2], ethers.zeroPadValue(safeToL2SetupLibAddress, 32).slice(2)),
+                );
+                const preDelegateCallStorage = trace.structLogs[delegateCallIntoTheLib].storage;
+
+                // The SafeSetup event is emitted after the Safe is set up
+                // To get the storage snapshot after the Safe is set up, we need to find the LOG2 opcode with the topic input on the stack equal the SafeSetup event signature
+                const SAFE_SETUP_EVENT_SIGNATURE = ethers.id("SafeSetup(address,address[],uint256,address,address)");
+                const postSafeSetup = trace.structLogs.find(
+                    (log, index) =>
+                        log.op === "LOG2" &&
+                        log.stack[log.stack.length - 3] === SAFE_SETUP_EVENT_SIGNATURE.slice(2) &&
+                        index > delegateCallIntoTheLib,
+                );
+                if (!postSafeSetup) {
+                    throw new Error("No SafeSetup event");
+                }
+                const postSafeSetupStorage = postSafeSetup?.storage;
+
+                for (const [key, value] of Object.entries(postSafeSetupStorage)) {
+                    // The slot key 0 is the singleton storage slot, it must equal the L2 singleton address
+                    if (sameHexString(key, ethers.zeroPadValue("0x00", 32))) {
+                        expect(sameHexString(ethers.zeroPadValue(safeL2SingletonAddress, 32), value)).to.be.true;
+                    } else {
+                        // All other storage slots must be the same as before the DELEGATECALL
+                        if (key in preDelegateCallStorage) {
+                            expect(sameHexString(preDelegateCallStorage[key], value)).to.be.true;
+                        } else {
+                            // This special case is needed because the SafeToL2Setup library inherits the SafeStorage library
+                            // And that makes the tracer report all the storage slots in the SafeStorage library as well
+                            // Even though if they were not touched during the transaction
+                            expect(sameHexString(value, "0".repeat(64))).to.be.true;
```

Comment:
To be extra sure, should we check that `key` belongs to the slots we expect to be initialized this way?

Reply:
They're set to 0 anyway, I am unsure if that provide any additional guarantees.

We can also replicate the [test](https://github.com/safe-global/safe-smart-account/blob/main/test/libraries/Safe.spec.ts) from `SafeStorage` lib to check that it follows the expected storage layout.

Reply:
Added [here](https://github.com/safe-global/safe-smart-account/pull/759/commits/fa9e29f2d5352f7afac3c643541aee97f0e3d938). LMK how it looks

Reply:
LGTM.